File: app/src/lib/pose/kip-colors.ts
import type { KipName } from '$lib/pose/pose-chart-types';

/**
 * Canonical KIP palette for charts and video overlay.
 * Keep in sync with backend/src/pipeline/kip_colors.py (light-theme --chart-* tokens).
 */
export const KIP_COLORS_HEX: Record<KipName, string> = {
	INSIDE_KNEE: '#F54900',
	OUTSIDE_HIP: '#009689',
	HIP_HINGE: '#104E64',
	FRONT_KNEE: '#FFB900'
};

export function hexColorForKipName(key: KipName): string {
	return KIP_COLORS_HEX[key];
}

File: backend/src/pipeline/kip_colors.py
"""Canonical KIP palette for video overlay (OpenCV BGR).

Keep in sync with app/src/lib/pose/kip-colors.ts (light-theme --chart-* tokens).
"""

from __future__ import annotations

KIP_COLORS_BGR: dict[str, tuple[int, int, int]] = {
    "INSIDE_KNEE": (0, 73, 245),
    "OUTSIDE_HIP": (137, 150, 0),
    "HIP_HINGE": (100, 78, 16),
    "FRONT_KNEE": (0, 185, 255),
}

KIP_LABELS: dict[str, str] = {
    "INSIDE_KNEE": "Inside Knee",
    "OUTSIDE_HIP": "Outside Hip",
    "HIP_HINGE": "Hip Hinge",
    "FRONT_KNEE": "Front Knee",
}
